libdebug/libdebug.py

```diff
@@ -889,6 +889,31 @@ def migrate_to_gdb(self, open_in_new_process: bool = True):
         # TODO: once we have signal handling, we should remove this
         self.step()
 
+    def finish(
+        self,
+        thread: ThreadContext | None = None,
+        exact: bool = True
+    ):
+        """Continues the process until the current function returns or the process stops. When used in step mode,
+        it will step until a return instruction is executed. Otherwise, it uses a heuristic
+        based on the call stack to breakpoint (exact is slower).
+        
+        Args:
+            thread (ThreadContext, optional): The thread to affect. Defaults to None.
+            exact (bool, optional): Whether or not to execute in step mode. Defaults to True.
+        """
+        self._ensure_process_stopped()
+
+        if thread is None:
+            # If no thread is specified, we use the first thread
+            thread = self.threads[0]
+
+        self._polling_thread_command_queue.put((self.__threaded_finish, (thread,exact)))
+
+        # Wait for the background thread to signal "task done" before returning
+        # We don't want any asynchronous behaviour here
+        self._polling_thread_command_queue.join()
```

Comment:
The syscall-hooking branch, which was recently merged into the dev branch, includes checks on the return values of the polling_thread operations to handle exceptions. These checks are necessary to prevent exceptions raised in the background thread from escaping the try-catch statements in the main thread, which could terminate the main user script. I would recommend adding the check here as well.
See https://github.com/libdebug/libdebug/pull/11#issue-2266625836

Reply:
On it!